app/services/fixed_charges/override_service.rb

```diff
@@ -35,7 +35,7 @@ def call
         FixedCharges::EmitEventsForActiveSubscriptionsService.call!(
           fixed_charge: new_fixed_charge,
           subscription:,
-          apply_units_immediately: true
+          apply_units_immediately: !!params[:apply_units_immediately]
```

Comment:
I am thinking:
will it then emit events when creating a subscription with overrides from the beginning and not sending :apply_units_imediately at all?

Reply:
good point, I'll ensure both scenarios (new subs and update sub) are covered by tests. Thanks for raising it.